lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +681,144 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    private final StatusEvent updateEventForDeletion(final StatusEvent ev)
+              throws RetriableIndexingException,
+                     IndexingException {
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        final int wsid = ev.getAccessGroupId().get();
+        final long objid = Long.valueOf(ev.getAccessGroupObjectId().get()).longValue();
+
+        // check if workspace is permanently deleted or marked as deleted
+        try {
+            getWorkspaceInfo(wsid);
```

Comment:
it occurs to me now that this block isn't really necessary - just try to listObjects and check the error message for a deleted workspace if that fails:

```
In [2]: from biokbase.workspace.client import Workspace
In [3]: ws = Workspace('https://ci.kbase.us/services/ws', token=gapricetoken)

In [4]: ws.create_workspace({'workspace': 'searchysearch'})
Out[4]: 
[31467,
 u'searchysearch',
 u'gaprice',
 u'2018-04-18T21:59:03+0000',
 0,
 u'a',
 u'n',
 u'unlocked',
 {}]

In [5]: ws.delete_workspace({'id': 31467})

In [6]: ws.list_objects({'ids': [31467]})
---------------------------------------------------------------------------
ServerError                               Traceback (most recent call last)
<ipython-input-6-268165ad2bd0> in <module>()
----> 1 ws.list_objects({'ids': [31467]})
*snip*

ServerError: JSONRPCError: -32500. Workspace 31467 is deleted
```

Reply:
That also eliminates the possibility of a race condition

Reply:
Done.